File: WebClient/src/Request.ts
import { IRequest } from "./IRequest";

export class Request implements IRequest {

    private correlationId: string;
    private content: string;
    private sender: string;
    private topic: string;
    private timeStamp: Date;
    private destination: string;

    constructor(id: string, content: string, sender: string, topic: string, destination: string) {
        this.correlationId = id;
        this.content = content;
        this.sender = sender;
        this.topic = topic;
        this.timeStamp = new Date();
        this.destination = destination;
    }

    public get Topic() { return this.topic; }
    public set Topic(value: string) { this.topic = value; }

    get CorrelationId() { return this.correlationId; }
    set CorrelationId(value: string) { this.correlationId = value; }

    get Sender() { return this.sender; }
    set Sender(value: string) { this.sender = value; }

    get Content() { return this.content; }
    set Content(value: string) { this.content = value; }

    get TimeStamp() { return this.timeStamp; }
    set TimeStamp(value: Date) { this.timeStamp = value; }

    get Destination() { return this.destination; }
    set Destination(value: string) { this.destination = value; }

}

File: Shared/Request.cs
using System;

using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Newtonsoft.Json;

namespace SignalRCommunicator
{
    [Serializable]
    class Request : IRequest
    {
        [JsonProperty]
        public string Responder { get; set; }

        [JsonProperty]
        public string CorrelationId { get; set; }

        [JsonProperty]
        public string Content { get; set; }

        [JsonProperty]
        public string Sender { get; set; }

        [JsonProperty]
        public DateTime Timestamp { get; set; }

        [JsonProperty]
        public string Topic { get; set; }

        public Request()
        {
            CorrelationId = Guid.NewGuid().ToString();
        }

        public Request(Message message, string responder)
            : this(responder, message.CorrelationId, message.Content, message.Sender, message.Timestamp)
        {

        }

        public Request(string responder, string content, string sender)
            :this()
        {
            Responder = responder;
            Content = content;
            Sender = sender;
            Timestamp = DateTime.Now;
        }

        public Request(
            string responder,
            string correlationId,
            string content,
            string sender,
            DateTime timestamp) : this()
        {
            Responder = responder;
            CorrelationId = correlationId;
            Content = content;
            Sender = sender;
            Timestamp = timestamp;
        }
    }
}